### PR TITLE
More tidying up of accounts 

### PR DESCRIPTION
**Pull request by hadley**
Explanations in individual commits

### CODE REVIEW

File: R/accounts.R [hunk 1]
```diff
@@ -60,74 +60,121 @@ connectApiUser <- function(account = NULL, server = NULL, apiKey, quiet = FALSE)
 
   registerAccount(
     serverName = server,
-    accountName = account %||% user$username,
+    accountName = user$username,
```
Comment (aronatkins) [inline R/accounts.R:63]: This change means that the `account` argument is unnecessary going forward. Something for the list.

File: R/accounts.R [hunk 2]
```diff
@@ -60,74 +60,121 @@ connectApiUser <- function(account = NULL, server = NULL, apiKey, quiet = FALSE)
...
     accountId = user$id,
     apiKey = apiKey
   )
 
   if (!quiet) {
-    message("Account registered successfully: ", account %||% user$username)
+    accountId <- accountId(user$username, server)
```
Comment (aronatkins) [inline R/accounts.R:69]: nit: There are two notions of `accountId`: the user identifier from the API, and `username@server`. That's bound to confuse our future selves.

File: R/accounts.R [hunk 3]
```diff
@@ -60,74 +60,121 @@ connectApiUser <- function(account = NULL, server = NULL, apiKey, quiet = FALSE)
...
-  repeat {
-    Sys.sleep(1)
-    user <- getAuthedUser(server, token = token)
-    if (!is.null(user))
-      break
+  if (isFALSE(launch.browser)) {
+    cli::cli_alert_warning("Open {.url {claim_url}} to authenticate")
+  } else {
+    cli::cli_alert_info("A browser window should open to complete authentication")
+    cli::cli_alert_warning("If doesn't open, then visit {.url {claim_url}}")
```
Comment (aronatkins) [inline R/accounts.R:122]: "if _ONE_ doesn't open"

File: R/accounts.R [hunk 4]
```diff
@@ -60,74 +60,121 @@ connectApiUser <- function(account = NULL, server = NULL, apiKey, quiet = FALSE)
...
-                               user$username, "'): "))
-      if (nchar(input) > 0)
-        user$username <- input
+  # keep trying to authenticate until we're successful; server returns
+  # 500 "Token is unclaimed error" while waiting for interactive auth to complete
+  cli::cli_progress_bar(format = "{cli::pb_spin} Waiting for authentication...")
+  repeat {
+    for (i in 1:10) {
+      Sys.sleep(0.1)
+      cli::cli_progress_update()
```
Comment (aronatkins) [inline R/accounts.R:131]: Why 10 progress updates between attempts? Maybe one progress update per attempt with ~1s between attempts?

File: R/rsa.R
Comment (aronatkins) [inline R/rsa.R:1]: With `generateToken` removed, there's nothing "RSA" about this file... the remaining `createUniqueId` exists only to assist `generateToken` and probably should float with it.

**Review by aronatkins — APPROVED** [review]

File: R/accounts.R (hunk 4 above)
Comment (hadley) [reply R/accounts.R:131]: Because then you get a cool spinner 😄

File: R/rsa.R
Comment (hadley) [reply R/rsa.R:1]: Ah, I missed that. And what do you think about replacing `createUniqueId(16)` with `paste(openssl::rand_bytes(16), collapse = "")` ?

Comment (hadley) [reply R/rsa.R:1]: (FWIW I'll probably split `accounts.R` into multiple pieces in a future PR)

File: R/accounts.R (hunk 1 above)
Comment (hadley) [reply R/accounts.R:63]: Hmmm, maybe I should leave as is for now. If we fix, I think it should be part of a general re-thinking of these functions (e.g. more consistent function names, argument names, and defaults)

File: R/accounts.R (hunk 2 above)
Comment (hadley) [reply R/accounts.R:69]: Yeah, I'll brainstorm a better name.

Comment (hadley) [reply R/accounts.R:69]: I have failed to come up with a better name; but I've made a note to keep thinking about it.

File: R/rsa.R
Comment (aronatkins) [reply R/rsa.R:1]: Any random data is fine. Watch out for R environments that might force their global random seed to a fixed state, as that means our "random token" ends up not-so-random.

Prior work:
https://github.com/rstudio/rsconnect/issues/221 / https://github.com/rstudio/rsconnect/pull/222
https://github.com/rstudio/rsconnect/issues/378 / https://github.com/rstudio/rsconnect/pull/379

Comment (hadley) [reply R/rsa.R:1]: Using openssl for this means that you get cryptographical secure random numbers, which is probably not important here, but they definitely affected by R's RNG state.

Comment (aronatkins) [reply R/rsa.R:1]: Did you mean "unaffected"? I'm +1 as long as caller cannot initialize (R or openssl) seed state to produce predictable identifiers.

**Pull request merged**